### PR TITLE
fix(migrations) postgres migrations tuning and json->jsonb, timestamp->timestamptz

### PR DESCRIPTION
**Pull request by bungle**
### Summary

- ~This PR changes base migration of `plugins` table `config` column type from `JSON` to `JSONB` (it was already converted on 0.14.0 to 0.15.0 migration)~
- ~It also changes `upstreams` table `healtchecks` column type from `JSON` to `JSONB` and adds `0.14.0` to `0.15.0` migration for the column type.~
- This PR changes `JSON` types to `JSONB` types on `0.14.0` to `0.15.0` migrations
- This PR changes `TIMESTAMP WITHOUT TIMEZONE` types to `TIMESTAMP WITH TIME ZONE` types on `0.14.0` to `0.15.0` migrations
- This PR cleans cruft left behind from **pre** `0.15.0` migrations to match the new database created with `kong migrations bootstrap && kong migrations up && kong migrations finish` on `0.15.0`.

### CODE REVIEW

File: kong/db/migrations/core/000_base.lua [hunk 1]
```diff
@@ -152,7 +152,7 @@ return {
         "hash_on_cookie"        TEXT,
         "hash_on_cookie_path"   TEXT,
         "slots"                 INTEGER                      NOT NULL,
-        "healthchecks"          JSON
+        "healthchecks"          JSONB
```
Comment (thibaultcha) [inline kong/db/migrations/core/000_base.lua]: This migration should truly reflect the state of the database as of 0.14, and we should not diverge away from it. The `14_to_15` migration will still run for freshly bootstrapped databases as well, so there is no need for this change anyway?

File: kong/db/migrations/core/001_14_to_15.lua [hunk 2]
```diff
@@ -33,6 +33,10 @@ return {
       $$;
 
       ALTER TABLE IF EXISTS ONLY "plugins" ALTER "config" TYPE JSONB USING "config"::JSONB;
+
+
+
+      ALTER TABLE IF EXISTS ONLY "upstreams" ALTER "healthchecks" TYPE JSONB USING "healthchecks"::JSONB;
```
Comment (thibaultcha) [inline kong/db/migrations/core/001_14_to_15.lua]: Out of curiosity: do you know what happens _if_ (big if) the type of this column is already `JSONB`?

Comment (bungle) [reply kong/db/migrations/core/001_14_to_15.lua]: I am not sure what happens internally, but I guess pretty much nothing happens. I tried it and, well nothing happened.

File: kong/db/migrations/core/000_base.lua (hunk 1 above)
Comment (bungle) [reply kong/db/migrations/core/000_base.lua]: Thanks for for this. I fully 100% agree.

File: kong/plugins/rate-limiting/migrations/000_base_rate_limiting.lua [hunk 3]
```diff
@@ -108,7 +108,7 @@ return {
         period      text,
         identifier  text,
         value       counter,
-        PRIMARY KEY ((route_id, service_id, api_id, identifier, period_date, period))
```
Comment (thibaultcha) [inline kong/plugins/rate-limiting/migrations/000_base_rate_limiting.lua]: Why removing those? They serve the purpose of specifying a compound partition key, and are essential. I recall similar attempt to remove them already in the past - unfortunately we cannot add comments in this migration...

File: kong/plugins/rate-limiting/migrations/001_14_to_15.lua [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+        ALTER "period_date" TYPE TIMESTAMP WITH TIME ZONE USING "period_date" AT TIME ZONE 'UTC';
+    ]],
+  },
+
+  cassandra = {
+    up = [[
+      BEGIN BATCH
+        -- nothing to migrate
+      APPLY BATCH;
+    ]],
```
Comment (thibaultcha) [inline kong/plugins/rate-limiting/migrations/001_14_to_15.lua]: It should be possible (and preferable) to specify an empty string for these NOP migrations; have you tried?

File: kong/plugins/rate-limiting/migrations/000_base_rate_limiting.lua (hunk 3 above)
Comment (bungle) [reply kong/plugins/rate-limiting/migrations/000_base_rate_limiting.lua]: Oh, it really looked like a tupo as it wasn't found anywhere else. I even tried to find this on C* docs, but couldn't. I should have tried harder. Thanks for educating me on this. I will fix it. Siirry.

File: kong/plugins/rate-limiting/migrations/001_14_to_15.lua (hunk 4 above)
Comment (bungle) [reply kong/plugins/rate-limiting/migrations/001_14_to_15.lua]: I tried, but it gave an error. We can fix why it errored, or keep this.

Comment (thibaultcha) [reply kong/plugins/rate-limiting/migrations/001_14_to_15.lua]: Could we fix the error then? Specifying the migration is required for the sake of explicitness as per [the migrations schema](https://github.com/Kong/kong/blob/next/kong/db/schema/others/migrations.lua#L11-L12), but the connectors should just NOP on empty migrations

Comment (bungle) [reply kong/plugins/rate-limiting/migrations/001_14_to_15.lua]: @thibaultcha I added `len_min = 0` to migrations schema.

**Pull request merged**